contracts/lbp/LBPManager.sol

```diff
@@ -105,66 +112,74 @@ contract LBPManager {
         uint256[] memory _endWeights,
         uint256[] memory _fees,
         bytes memory _metadata
-    ) external returns (address) {
+    ) external {
         require(!initialized, "LBPManager: already initialized");
         require(_beneficiary != address(0), "LBPManager: _beneficiary is zero");
 
         initialized = true;
         admin = msg.sender;
+        swapFeePercentage = _fees[0];
         feePercentage = _fees[1];
         beneficiary = _beneficiary;
         metadata = _metadata;
+        startTimeEndTime = _startTimeEndTime;
+        name = _name;
+        symbol = _symbol;
+        LBPFactory = _LBPFactory;
 
         // Token addresses are sorted in numerical order (ascending) as specified by Balancer
         if (address(_tokenList[0]) > address(_tokenList[1])) {
+            projectTokenIndex = 1;
             tokenList.push(_tokenList[1]);
             tokenList.push(_tokenList[0]);
+
             amounts.push(_amounts[1]);
             amounts.push(_amounts[0]);
 
-            projectTokenIndex = 1;
-            uint256 swap = _startWeights[0];
-            _startWeights[0] = _startWeights[1];
-            _startWeights[1] = swap;
+            startWeights.push(_startWeights[1]);
+            startWeights.push(_startWeights[0]);
 
-            swap = _endWeights[0];
-            _endWeights[0] = _endWeights[1];
-            _endWeights[1] = swap;
+            endWeights.push(_endWeights[1]);
+            endWeights.push(_endWeights[0]);
         } else {
             projectTokenIndex = 0;
             tokenList = _tokenList;
             amounts = _amounts;
+            startWeights = _startWeights;
+            endWeights = _endWeights;
         }
+    }
+
+    /**
+     * @dev                             Subtracts the fee and adds liquidity to the LBP.
```

Comment:
I think this is of.

Reply:
Sry, I checked it on my phone and only read `Subtracts the fee`. I would still extend it:
```suggestion
     * @dev                             Subtracts the fee, deploys the LBP and adds liquidity to it.
```